examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/SelectActionFragment.kt

```diff
@@ -168,7 +168,7 @@ class SelectActionFragment : Fragment() {
    * Notifies listener of Scan QR code button click.
    */
   private fun handleScanQrCodeClicked() {
-    showFragment(BarcodeFragment.newInstance(), false)
+    showFragment(BarcodeFragment.newInstance())
```

Comment:
What is the crash? why fragmentTransaction.addToBackStack(null) can solve the issue?

Reply:
I've (quickly) tested this and it resolves the crashing issue.

I believe what was happening that if you didn't add it to the backstack, when you get to the QR scanning camera, it will scan the QR code, and then try to go back to the host fragment. But because the fragment wasn't on the backstack, it returns as null, and doesn't know where to return to.

Logcat with some debug in:
[logcat_crash.log](https://github.com/project-chip/connectedhomeip/files/12037141/logcat_crash.log)

If you then remove the false value so that the fragment goes on the backstack, when you do getHost, it passes in a valid Fragment that you can determine the activity to associate with it.
[logcat_no_crash.log](https://github.com/project-chip/connectedhomeip/files/12037171/logcat_no_crash.log)

For reference the debug I put in was
  fun <T> getHost(source: Fragment, hostClass: Class<T>): T? {
    val activity = source.activity
    val parentFragment = source.parentFragment
    Log.d("Luke", "activity - " + activity?.let { it::class.java.simpleName } + ". Parent Fragment - " + parentFragment?.let { it::class.java.simpleName })

Reply:
during my local test, I observer the similar issue, and find this analysis aligns with my observation. Considering multiple companies encounter the similar issue, we would go ahead to merge it, thanks